### PR TITLE
Update TabViewItem to properly support some properties

### PR DESCRIPTION
**Pull request by gabrielconl**
TabViewItem and its overlay corners are updated to work properly with the following properties: CornerRadius, Background, BorderBrush, BorderThickness.

Changes:
- the triangle and arc for each overlay are grouped
- the tab background is separated from the tab container
- added a filler background that joins the overlays and the tab background
- the tab content was separated into the actual tab and the separator

### CODE REVIEW

**Review by gabrielconl — COMMENTED** [review]
*_Make bottom corners get updated_, wanted to say "allow to" first time 😕

File: dev/TabView/TabView.xaml [hunk 1]
```diff
@@ -704,6 +677,44 @@
...
+                            Data="M0 0C0 1.19469 0.523755 2.26706 1.35418 3H4C2.34315 3 1 1.65685 1 0H0Z"/>
+
+                        <Path x:Name="RightRadiusRenderTriangle"
+                            x:Load="False"
+                            Grid.Column="2"
+                            Visibility="Collapsed"
+                            Fill="{ThemeResource TabViewItemHeaderBackgroundSelected}"
+                            VerticalAlignment="Bottom"
+                            Margin="0,0,-4,0"
+                            Data="M4 4H0V0C0 2.20914 1.79086 4 4 4Z"/>
```
Comment (chingucoding) [inline dev/TabView/TabView.xaml:717]: Why did you move those lines?

File: dev/TabView/TabView_themeresources.xaml [hunk 2]
```diff
@@ -60,15 +60,7 @@
...
-            <LinearGradientBrush x:Key="TabViewSelectedItemBorderBrush" MappingMode="Absolute" StartPoint="0,0" EndPoint="0,4">
-                <LinearGradientBrush.RelativeTransform>
-                    <ScaleTransform ScaleY="-1" CenterY="0.5"/>
-                </LinearGradientBrush.RelativeTransform>
-                <LinearGradientBrush.GradientStops>
-                    <GradientStop Offset="1.0" Color="Transparent"/>
-                    <GradientStop Offset="1.0" Color="{ThemeResource CardStrokeColorDefault}"/>
-                </LinearGradientBrush.GradientStops>
-            </LinearGradientBrush>
+            <StaticResource x:Key="TabViewSelectedItemBorderBrush"                          ResourceKey="CardStrokeColorDefault" />
```
Comment (chingucoding) [inline dev/TabView/TabView_themeresources.xaml:63]: What is the reason for that change? I don't think that changing this resource makes sense in this PR.

Comment (gabrielconl) [reply dev/TabView/TabView_themeresources.xaml:63]: image
This and the other change were done because the corners were rendered behind the tab before.
The tab container used a gradient that became transparent where the corners would be, but now you can set `TabViewSelectedItemBorderBrush` and affect the entire tab.

Comment (beervoley) [reply dev/TabView/TabView_themeresources.xaml:63]: This way you beat the purpose of having the "arcs". The gradient exists to make smooth transition from the TabViewItem to the arcs on the sides. If you use SolidColorBrush instead of gradient, you will end up with 
image

it just beats the purpose of having smooth arcs on the sides:)

Comment (gabrielconl) [reply dev/TabView/TabView_themeresources.xaml:63]: That image was meant to show what happens if the arcs are left before the tab container. This is how it works with my change:
image

You can see a bit of the tab border at the bottom because of #5099, but if you used a single color gradient it would look normal.
image

Comment (beervoley) [reply dev/TabView/TabView_themeresources.xaml:63]: this will only work if the background is solid of both the arcs and the and the tabviewitem :)
if they're translucent, you'll see that little piece of tab border in the bottom anyway.
I see what you mean, this might be a good idea, we're using solid colors anyway
And this way we don't have to use the gradient I suppose?

File: dev/TabView/TabView.xaml (hunk 1 above)
Comment (beervoley) [reply dev/TabView/TabView.xaml:717]: @chingucoding to give the arcs bigger zIndex so they can draw on top of the tab container :)

File: dev/TabView/TabView_themeresources.xaml (hunk 2 above)
Comment (gabrielconl) [reply dev/TabView/TabView_themeresources.xaml:63]: Yes, removing the gradient was so that you could use a single brush for both the arcs and the container. But I agree that it's not a good option with a semi-transparent tab.

File: dev/TabView/TabView.xaml (hunk 1 above)
Comment (chingucoding) [reply dev/TabView/TabView.xaml:717]: Did it have to draw on top of the container? I thought we moved it out wide enough.

Comment (beervoley) [reply dev/TabView/TabView.xaml:717]: @chingucoding with this PR's proposal - we want to remove the gradient on the stroke color, meaning that the Tab border will go all the way to the bottom instead of nicely connecting at 4px height with the arcs on the side. In order to mitigate this, we can draw arks on top of the container so the arks themselves will cover the bottom part of the border on the sides so it's not visible:)

Comment (chingucoding) [reply dev/TabView/TabView.xaml:717]: I see, that makes sense.

**Comment by mdtauk** [comment]
In with the obligitory, will these Tabs and TabBar work when using AcrylicBrushes, InApp and HostBackdrop? @chingucoding

**Comment by gabrielconl** [comment]
I found another solution that works with transparency.
It basically adds another grid for the border with a bottom margin.
image

@mdtauk it works really well with acrylic
image

**Comment by gabrielconl** [comment]
I have a larger commit that changes almost everything in this PR and adds support for more properties. Should I scrap this PR and do a new one?

**Comment by chingucoding** [comment]
What do you mean with support for more properties?

**Comment by gabrielconl** [comment]
> What do you mean with support for more properties?

I got the corners to also support BorderThickness and CornerRadius. This required a slightly larger change, but I think it still fits in this PR.

**Comment by chingucoding** [comment]
> I got the corners to also support BorderThickness and CornerRadius. This required a slightly larger change, but I think it still fits in this PR.

I think it might be good to have that in a single PR, especially if they touch the same file.

**Comment by gabrielconl** [comment]
It's quite a big change, but I think it's worth it. I still have to work on some things though.
The default look should be the same.

**Comment by ranjeshj** [comment]
@gabrielconl For larger changes can you please open an issue/discuss go through the contribution guidelines/process outlined here - https://github.com/microsoft/microsoft-ui-xaml/blob/main/docs/contribution_workflow.md#contribution-workflow

**Comment by gabrielconl** [comment]
Ah, sorry for that. It started as a small change and I kept updating it to work with more things, it didn't really cross my mind to create an issue.

**Comment by gabrielconl** [comment]
Since this needs more discussion and the PR will possibly be scrapped, I'll close it for now.
